### PR TITLE
CLOUDP-244165: remove ResourceWatcher in AtlasProject

### PR DESCRIPTION
**Pull request by s-urbaniak**
### All Submissions:

* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/indexer/atlasprojectsecrets.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		a.logger.Errorf("expected *akov2.AtlasProject but got %T", object)
+		return nil
+	}
+
+	result := sets.New[string]()
+	addIfNotEmpty := func(ref *common.ResourceRefNamespaced) {
+		if !ref.IsEmpty() {
+			result.Insert(ref.GetObject(project.Namespace).String())
+		}
+	}
```
Comment (josvazg) [inline pkg/indexer/atlasprojectsecrets.go:46]: nitpick

Why is this not a regular private function?

```go
	func addIfNotEmpty(result sets.Set, ref *common.ResourceRefNamespaced) {
		if !ref.IsEmpty() {
			result.Insert(ref.GetObject(project.Namespace).String())
		}
	}
```

I do not see the reason for this to be a closure. Am I missing something?

**Review by josvazg — APPROVED** [review]
LGTM! Just a small question/nitpick

File: pkg/indexer/atlasprojectsecrets.go (hunk 1 above)
Comment (s-urbaniak) [reply pkg/indexer/atlasprojectsecrets.go:46]: yes, this is enclosing over `result`

Comment (s-urbaniak) [reply pkg/indexer/atlasprojectsecrets.go:46]: but if you find
```
func addIfNotEmpty(set sets.Set[string], parentNamespace string, ref *common.ResourceRefNamespaced) {
	if !ref.IsEmpty() {
		set.Insert(ref.GetObject(parentNamespace).String())
	}
}
```
and
```
			addIfNotEmpty(result, project.Namespace, &notification.APITokenRef)
			addIfNotEmpty(result, project.Namespace, &notification.DatadogAPIKeyRef)
			addIfNotEmpty(result, project.Namespace, &notification.FlowdockAPITokenRef)
			addIfNotEmpty(result, project.Namespace, &notification.OpsGenieAPIKeyRef)
			addIfNotEmpty(result, project.Namespace, &notification.ServiceKeyRef)
			addIfNotEmpty(result, project.Namespace, &notification.VictorOpsSecretRef)
```

I can change it

Comment (josvazg) [reply pkg/indexer/atlasprojectsecrets.go:46]: meh, not a big deal

**Review by helderjs — APPROVED** [review]

**Pull request merged**